usr/lib/cca_stdll/cca_specific.c

```diff
@@ -1206,6 +1208,288 @@ CK_RV token_specific_rsa_verify(STDLL_TokData_t * tokdata,
     return CKR_OK;
 }
 
+CK_RV token_specific_rsa_pss_sign(STDLL_TokData_t *tokdata,
+                                  SESSION  *sess,
+                                  SIGN_VERIFY_CONTEXT *ctx,
+                                  CK_BYTE *in_data,
+                                  CK_ULONG in_data_len,
+                                  CK_BYTE *out_data,
+                                  CK_ULONG *out_data_len)
+{
+    CK_RSA_PKCS_PSS_PARAMS *pss;
+    long return_code, reason_code, rule_array_count;
+    unsigned char rule_array[CCA_RULE_ARRAY_SIZE] = { 0, };
+    long signature_bit_length, message_len;
+    CK_ATTRIBUTE *attr;
+    OBJECT *key_obj = NULL;
+    CK_BYTE *message = NULL;
+    CK_RV rc;
+
+    UNUSED(tokdata);
+    UNUSED(sess);
+
+    rc = object_mgr_find_in_map1(tokdata, ctx->key, &key_obj, READ_LOCK);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("object_mgr_find_in_map1 failed\n");
+        goto done;
```

Comment:
sure you want to go down and do a object_put() here ?
I would recommend to return rc here.

Reply:
Yes we could got out here directly, but I prefer to always use the out label for all cases. Makes it easier to add something above in the future, or also add exit traces at the end of the routine.